apollo-router/src/plugins/telemetry/config_new/extendable.rs

```diff
@@ -255,6 +256,23 @@ where
     }
 }
 
+impl<A, E, Request, Response, EventResponse> Extendable<A, E>
+where
+    A: Default + Selectors<Request = Request, Response = Response, EventResponse = EventResponse>,
+    E: Selector<Request = Request, Response = Response, EventResponse = EventResponse>,
+{
+    pub(crate) fn validate(&self, restricted_stage: Option<Stage>) -> Result<(), String> {
+        if let Some(Stage::Request) = &restricted_stage {
+            for (name, custom) in &self.custom {
+                if !custom.is_active(Stage::Request) {
+                    return Err(format!("cannot set the attribute {name:?} because it's using a selector computed in another stage than 'request' because it won't be computed"));
```

Comment:
```suggestion
                    return Err(format!("cannot set the attribute {name:?} because it is using a selector computed in another stage than 'request' so it will not be computed"));
```